aquacomputer_d5next.c

```diff
@@ -1121,44 +1247,77 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 			}
 
 			/* Decrement to convert from hwmon to aqc */
-			ret = aqc_set_ctrl_val(priv, priv->fan_ctrl_offsets[channel], val - 1, 8);
+			ret = aqc_set_ctrl_val(priv,
+					       priv->fan_ctrl_offsets[channel], val - 1, AQC_8);
 			if (ret < 0)
 				return ret;
 			break;
 		case hwmon_pwm_input:
-			pwm_value = aqc_pwm_to_percent(val);
-			if (pwm_value < 0)
-				return pwm_value;
-			if (priv->kind == aquaero) {
+			if (val < 0 || val > 255)
```

Comment:
Please move this check back to aqc_pwm_to_percent().

Reply:
If I move that check back to aqc_pwm_to_percent(), it does not cover the aquastreamxt case anymore. Then we need to add the same check in another place just for the aquastream. So I would like to leave it as is.